### PR TITLE
Export layer lmdb replay log size, et al

### PR DESCRIPTION
**Pull request by jnwatson**
SYN_SLAB_COMMIT_PERIOD may be used to set the commit period.  Don't mess with this unless you know what you're doing.

Add layer:lmdb:max_replay_log setting to cortex.

Fix the way that lmdb:growsize in layer was provided to lmdbslab.

### CODE REVIEW

File: synapse/lib/layer.py [hunk 1]
```diff
@@ -1451,25 +1451,29 @@ async def _layrV5toV6(self):
...
             'lockmemory': self.lockmemory,
-            'growsize': self.growsize,
+            'map_async': self.mapasync,
         }
 
+        if self.growsize is not None:
+            slabopts['growsize'] = self.growsize
+
+        otherslabopts = slabopts.copy()
+        otherslabopts['readahead'] = False
```
Comment (invisig0th) [inline synapse/lib/layer.py:1470]: Why is this the new default and what effect will it have on existing deployments?

File: synapse/lib/lmdbslab.py [hunk 2]
```diff
@@ -573,7 +573,7 @@ class Slab(s_base.Base):
     synctask = None
     syncevnt = None  # set this event to trigger a sync
 
-    COMMIT_PERIOD = 0.2  # time between commits
+    COMMIT_PERIOD = os.environ.get('SYN_SLAB_COMMIT_PERIOD', 0.2)  # time between commits
```
Comment (invisig0th) [inline synapse/lib/lmdbslab.py:576]: Env vars are strings, does this get normalized or cast somewhere?

File: synapse/cortex.py [hunk 3]
```diff
@@ -935,6 +934,11 @@ class Cortex(s_cell.Cell):  # type: ignore
             'description': 'Set the default lmdb:map_async value in LMDB layers.',
             'type': 'boolean'
         },
+        'layer:lmdb:max_replay_log': {
+            'default': True,
+            'description': 'Set the max size of the replay log for all layers.',
+            'type': 'boolean'
```
Comment (invisig0th) [inline synapse/cortex.py:940]: Should this be an int rather than a bool to make it actually tunable?

File: synapse/cortex.py [hunk 4]
```diff
@@ -3392,6 +3396,7 @@ async def addLayer(self, ldef=None, nexs=True):
         ldef.setdefault('creator', self.auth.rootuser.iden)
         ldef.setdefault('lockmemory', self.conf.get('layers:lockmemory'))
         ldef.setdefault('logedits', self.conf.get('layers:logedits'))
+        ldef.setdefault('logedits', self.conf.get('layers:logedits'))
```
Comment (vEpiphyte) [inline synapse/cortex.py:3399]: duplicate line - remove it

File: synapse/lib/layer.py [hunk 5]
```diff
@@ -1101,7 +1101,7 @@ class Layer(s_nexus.Pusher):
     def __repr__(self):
         return f'Layer ({self.__class__.__name__}): {self.iden}'
 
-    async def __anit__(self, layrinfo, dirn, nexsroot=None, allow_upstream=True):
+    async def __anit__(self, layrinfo, dirn, nexsroot=None, allow_upstream=True, mapasync=True, maxreplaylog=10000):
```
Comment (invisig0th) [inline synapse/lib/layer.py:1104]: confused about bool config vs int use here?

**Review by invisig0th — DISMISSED** [review]

File: synapse/cortex.py (hunk 3 above)
Comment (vEpiphyte) [inline synapse/cortex.py:940]: The default arg for the layer anit is ``maxreplaylog=10000`` but this confdef is typed as a boolean? What is the expected behavior here?

File: synapse/lib/layer.py [hunk 6]
```diff
@@ -1451,25 +1451,29 @@ async def _layrV5toV6(self):
...
     async def _initLayerStorage(self):
 
         slabopts = {
             'readonly': self.readonly,
             'max_dbs': 128,
-            'map_async': True,
             'readahead': True,
             'lockmemory': self.lockmemory,
-            'growsize': self.growsize,
+            'map_async': self.mapasync,
```
Comment (vEpiphyte) [inline synapse/lib/layer.py:1463]: Should ``self.maxreplaylog`` be mapped in here as ``max_replay_log`` ?

File: synapse/lib/layer.py (hunk 1 above)
Comment (jnwatson) [reply synapse/lib/layer.py:1470]: Ultimately this changes the nodeeditsslab mapasync to be consistent with the user setting (usually no changes), and turns off readahead for the nodeeditslab, which should slightly reduce I/O overhead for that. These are changes we probably wanted.

Restructured to be a bit more clear.

File: synapse/lib/lmdbslab.py (hunk 2 above)
Comment (jnwatson) [reply synapse/lib/lmdbslab.py:576]: Fixed.

File: synapse/cortex.py (hunk 3 above)
Comment (jnwatson) [reply synapse/cortex.py:940]: Yes

File: synapse/lib/layer.py (hunk 6 above)
Comment (jnwatson) [reply synapse/lib/layer.py:1463]: yes. sorry forgot to mark PR as WIP

File: synapse/tests/test_lib_lmdbslab.py [hunk 7]
```diff
@@ -334,7 +334,7 @@ async def test_lmdbslab_max_replay(self):
 
             # Make sure that we don't confuse the periodic commit with the max replay log commit
 
-            with patch('synapse.lib.lmdbslab.Slab.COMMIT_PERIOD', 10):
+            with patch.dict(os.environ, {"SYN_SLAB_COMMIT_PERIOD": "10"}):
```
Comment (vEpiphyte) [inline synapse/tests/test_lib_lmdbslab.py:337]: ``self.setTstEnvars`` should be used here to set a specific envar

Comment (jnwatson) [reply synapse/tests/test_lib_lmdbslab.py]: ok

**Review by vEpiphyte — APPROVED** [review]

**Pull request merged**